protocol/carbon/metricdeconstructor/configurableregexdeconstructor.go

```diff
@@ -0,0 +1,97 @@
+package metricdeconstructor
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/signalfx/golib/datapoint"
+	"regexp"
+	"strings"
+	"sort"
+)
+
+type configurableRegexMetricRule struct {
+	typer
+	Regex                string            `json:"Regex"`
+	MetricName           string            `json:"MetricName"`
+	AdditionalDimensions map[string]string `json:"AdditionalDimensions"`
+	MetricTypeString     string            `json:"MetricType"`
+	MetricType           datapoint.MetricType
+	Compiled             *regexp.Regexp
+	UseCount             int64
+}
+
+type configurableRegexMetricDeconstructor struct {
+	MetricRules                 []*configurableRegexMetricRule `json:"MetricRules"`
+	FallBackDeconstructorName   string                         `json:"FallbackDeconstructor"`
+	FallBackDeconstructorConfig string                         `json:"FallbackDeconstructorConfig"`
+	AdditionalDimensions        map[string]string              `json:"Dimensions"`
+	FallBackDeconstructor       MetricDeconstructor
+}
+
+func regexJSONLoader(config map[string]interface{}) (MetricDeconstructor, error) {
+	var metrics configurableRegexMetricDeconstructor
+	// a wee bit of a hack, but at least we know it's valid json and it makes the config easier
+	jsonString, _ := json.Marshal(config)
```

Comment:
I'm worried we may refactor the code and then this error would be ignored.  Generally, it's best practice in go to check and return all errors.  Can you check it, even if it's currently unable to throw?  You can trigger it like I mentioned above.